tests/unit/handlers/test__helpers.py

```diff
@@ -501,43 +507,62 @@ def test_no_span(self):
         self.assertEqual(sampled, False)
 
     def test_no_trace(self):
-        header = "/12345"
+        input_span = "12345"
+        expected_span = "3039".zfill(16)
+        header = f"/{input_span}"
         trace_id, span_id, sampled = self._call_fut(header)
         self.assertIsNone(trace_id)
-        self.assertEqual(span_id, "12345")
+        self.assertEqual(span_id, expected_span)
         self.assertEqual(sampled, False)
 
     def test_with_span(self):
         expected_trace = "12345"
-        expected_span = "67890"
-        header = f"{expected_trace}/{expected_span}"
+        input_span = "67890"
+        expected_span = "10932".zfill(16)
+        header = f"{expected_trace}/{input_span}"
         trace_id, span_id, sampled = self._call_fut(header)
         self.assertEqual(trace_id, expected_trace)
         self.assertEqual(span_id, expected_span)
         self.assertEqual(sampled, False)
 
+    def test_with_span_decimal_not_in_bounds(self):
+        input_spans = ["0", "9" * 100]
+
+        for input_span in input_spans:
+            expected_trace = "12345"
+            input_span = "67890"
+            expected_span = "10932".zfill(16)
```

Comment:
Why is this the expected span when the input span is invalid? Shouldn't the expected span be None?